x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -61,13 +103,21 @@ public DataFrameAnalyticsConfig(String id, String source, String dest, List<Data
         if (analyses.size() > 1) {
             throw new UnsupportedOperationException("Does not yet support multiple analyses");
         }
+        this.query = query == null ? null : Collections.unmodifiableMap(query);
+        this.querySupplier = new CachedSupplier<>(() -> lazyQueryParser.apply(query, id, new ArrayList<>()));
     }
 
     public DataFrameAnalyticsConfig(StreamInput in) throws IOException {
         id = in.readString();
         source = in.readString();
         dest = in.readString();
         analyses = in.readList(DataFrameAnalysisConfig::new);
+        if (in.readBoolean()) {
+            this.query = in.readMap();
+        } else {
+            this.query = null;
```

Comment:
If we default query to the match-all query, then we don't have to deal with `null` anywhere.